kernel/syscalls/kill.rs

```diff
@@ -0,0 +1,25 @@
+use crate::process::{current_process, signal::Signal, PId, Process};
+use crate::result::Errno;
+use crate::result::Result;
+use crate::syscalls::SyscallHandler;
+
+impl<'a> SyscallHandler<'a> {
+    pub fn sys_kill(&self, pid: PId, sig: Signal) -> Result<isize> {
+        let pid_int = pid.as_i32();
+        match pid_int {
+            pid_int if pid_int > 0 => match Process::find_by_pid(pid) {
+                Some(_) => Process::find_by_pid(pid).unwrap().send_signal(sig),
```

Comment:
```suggestion
                Some(proc) => proc.send_signal(sig),
```

I guess this is what you want to do.